### PR TITLE
refactor(api): common client for all API usage

### LINKED ISSUE #50: Abstract all API calls to `utils/api.js`
This way we can centralize logic for getting/posting to the backend, as well as centralize logic for handling errors.

### PR DESCRIPTION
**Pull request by williamtdr**
Resolves #50. Extends the API interface to cover all endpoints used by the frontend, and replaces all direct usages of endpoints (via `fetch`, etc) in the code with a call to the client. Also replaces the ticket listing view with _(No Recent TIckets_ if no jobs have yet been submitted.

Note there's many usages of the API that swallow errors without surfacing them to the user. They'll still appear on the console. I've marked these with TODO entries to come back to later.

### CODE REVIEW

File: src/utils/api.js [hunk 1]
```diff
@@ -1,37 +1,74 @@
...
-      console.error(res);
-      return null;
-    }
-    // we made it this far, good to return json
-    return await res.json();
-  } catch (error) {
-    // handle error
-    console.error(error);
-    return null;
+const API_STUB = 'http://localhost:8080/api';
```
[the remark this answers is not in the document]
Comment (williamtdr) [reply src/utils/api.js]: OK, cool!

File: src/utils/api.js [hunk 2]
```diff
@@ -1,37 +1,74 @@
...
+  return fetch(`${API_STUB}/tickets/${ticketID}`)
+    .then(throwIfNot200)
+    .then(convertToJSON)
+    .then(throwIfRepsonseNotSuccessful)
+    .catch(logOnFailure);
+};
+
+export const createTicket = body => {
+  return fetch(`${API_STUB}/tickets`, {
+    method: 'POST',
```
[the remark this answers is not in the document]
Comment (williamtdr) [reply src/utils/api.js]: :)

File: src/views/tickets/Ticket.js [hunk 3]
```diff
@@ -36,26 +37,19 @@ export default class Ticket extends Component {
   }
 
   refreshTicketState = async e => {
```
[the remark this answers is not in the document]
Comment (williamtdr) [reply src/views/tickets/Ticket.js]: Works foe m!

**Comment by tantric-slug** [comment]
Resolved conflicts with the fix/url-overflow that I just merged

**Comment by williamtdr** [comment]
@tantric-slug Thanks for doing that proactively!

File: src/views/tickets/Ticket.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (ThomasRiddle) [reply src/views/tickets/Ticket.js]: I tried to put the function into componentDidMount(), but I cannot load the existed ticket and there are no errors in the console. (on the branch "issue_49") I stuck here for long.

**Pull request merged**